components/portal/messages/controllers.js

```diff
@@ -34,36 +34,45 @@ define(['angular'], function(angular) {
         $scope.APP_FLAGS = APP_FLAGS;
         $scope.MISC_URLS = MISC_URLS;
         $scope.showMessagesFeatures = true;
+        $scope.seenMessageIds = [];
+        var seenIdOverride = false;
 
         // ////////////////
         // Local methods //
         // ////////////////
         /**
-         * Get all messages, then pass result on for filtering
+         * Get all messages, regardless of validity
```

Comment:
Maybe "validity" isn't the best way to characterize what this is regardless of? Validity is kind of about brokenness, as in invalid input, validating forms. Is this regardless of "applicability to the viewing user"?  Noting that this *is* clarified by the next couple lines of comment.